daemons/fenced/fenced_commands.c

```diff
@@ -2384,36 +2384,34 @@ send_async_reply(async_command_t *cmd, const pcmk__action_result_t *result,
                  int pid, bool merged)
 {
     xmlNode *reply = NULL;
-    bool bcast = false;
 
     CRM_CHECK((cmd != NULL) && (result != NULL), return);
 
+    log_async_result(cmd, result, pid, NULL, merged);
+
     reply = construct_async_reply(cmd, result);
+    if (merged) {
+        pcmk__xe_set_bool_attr(reply, F_STONITH_MERGED, true);
+    }
 
-    // If target was also the originator, broadcast fencing results for it
     if (!stand_alone && pcmk__is_fencing_action(cmd->action)
         && pcmk__str_eq(cmd->origin, cmd->victim, pcmk__str_casei)) {
-
+        /* The target was also the originator, so broadcast the result on its
+         * behalf (since it will be unable to).
+         */
         crm_trace("Broadcast '%s' result for %s (target was also originator)",
                   cmd->action, cmd->victim);
         crm_xml_add(reply, F_SUBTYPE, "broadcast");
         crm_xml_add(reply, F_STONITH_OPERATION, T_STONITH_NOTIFY);
-        bcast = true;
-    }
-
-    log_async_result(cmd, result, pid, NULL, merged);
-
-    if (merged) {
-        pcmk__xe_set_bool_attr(reply, F_STONITH_MERGED, true);
-    }
-    crm_log_xml_trace(reply, "Reply");
-
-    if (bcast) {
         send_cluster_message(NULL, crm_msg_stonith_ng, reply, FALSE);
     } else {
+        // Reply only to the originator
         stonith_send_reply(reply, cmd->options, cmd->origin, cmd->client);
     }
 
+    crm_log_xml_trace(reply, "Reply");
```

Comment:
exactly that was the log which I had meant to move to the end - which you did ;-)